### ISSUE TITLE
ClickHouse upgrade to latest version

### ISSUE THREAD

**Issue by guidoiaquinti** [PostHog/posthog#7028]
## Is your feature request related to a problem?
No

## Describe the solution you'd like
We would like to upgrade ClickHouse to the latest stable version (currently `v21.11.2.2-stable`). We are quite few versions behind and missing some important improvements like:

* arm64 support (important for development experience on Apple M1)
* general perf improvements
* improved window functions
* improved JSON handling functions
* we have some queries like _funnel time to convert_ that don't work on ClickHouse <= 21.9 (some int is not cast properly)

We should provide a battle tested upgrade path for both:
* PostHog cloud
* self-hosted PostHog installations

## Describe alternatives you've considered
Do nothing.

**Comment by macobo** [comment]
Related: https://github.com/PostHog/posthog/issues/5684

**Comment by tiina303** [comment]
TLDR from [slack thread](https://posthog.slack.com/archives/C01MM7VT7MG/p1636548564477500?thread_ts=1636547707.475200&cid=C01MM7VT7MG): Events table migration (pre-req for upgrade) is done for Cloud but not for self-hosted and that's tricky because we don't have a good story for migrations that can take days yet.

**Comment by mariusandra** [comment]
Related: https://github.com/PostHog/posthog/issues/6535

**Comment by guidoiaquinti** [comment]
Target release is ~`22.1`~ `22.3`

**Comment by neilkakkar** [comment]
Just a heads up: When we've figured out migration plans and everything, a final blocker to this upgrade is code: Some tests fail depending on the Clickhouse version, because indeterministic/bugs/etc. etc. So, alongside upgrading CH, we'd want to update some of this code as well.

**Comment by macobo** [comment]
We now support workflow_dispatch for the Backend CI where you can set the clickhouse version. If someone wants to dig into the failing tests, here they are: https://github.com/PostHog/posthog/actions/workflows/ci-backend.yml?query=event%3Aworkflow_dispatch

I gave the false impression that they all passed earlier - sorry! I thought they did :/

**Comment by guidoiaquinti** [comment]
This is now completed
